### ISSUE TITLE
Build warnings [-Wparentheses-equality] in thread code

### ISSUE THREAD

**Issue by NHDaly** [xianyi/OpenBLAS#1149]
Hi! :)

I saw these warnings when building [`julia`](https://github.com/JuliaLang/julia/), which uses OpenBLAS (via this import statement: [/deps/blas.mk#L3](https://github.com/JuliaLang/julia/blob/master/deps/blas.mk#L3)).

The following warning is repeated over and over, hundreds of times. Here is a snippet:

```
...
In file included from symm_k.c:77:
./level3_thread.c:295:17: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
  if ((alpha[0] == ZERO)
       ~~~~~~~~~^~~~~~~
./level3_thread.c:295:17: note: remove extraneous parentheses around the comparison to silence this warning
  if ((alpha[0] == ZERO)
      ~         ^      ~
./level3_thread.c:295:17: note: use '=' to turn this equality comparison into an assignment
  if ((alpha[0] == ZERO)
                ^~
                =
1 warning generated.
1 warning generated.
In file included from syrk_k.c:102:
./level3_syrk_threaded.c:203:17: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
  if ((alpha[0] == ZERO)
       ~~~~~~~~~^~~~~~~
./level3_syrk_threaded.c:203:17: note: remove extraneous parentheses around the comparison to silence this warning
  if ((alpha[0] == ZERO)
      ~         ^      ~
./level3_syrk_threaded.c:203:17: note: use '=' to turn this equality comparison into an assignment
  if ((alpha[0] == ZERO)
                ^~
                =
...
```

It looks like its referencing [these](https://github.com/xianyi/OpenBLAS/blob/b09cc3b9bbc5a66aed6cd514c676d043cddbd221/driver/level3/level3_syrk_threaded.c#L203) [comparison](https://github.com/xianyi/OpenBLAS/blob/b09cc3b9bbc5a66aed6cd514c676d043cddbd221/driver/level3/level3_thread.c#L296) [lines](https://github.com/xianyi/OpenBLAS/blob/b09cc3b9bbc5a66aed6cd514c676d043cddbd221/driver/level3/level3_syr2k.c#L157), which all use a macro like `#if defined(COMPLEX)` to guard out another check. So if `COMPLEX` isn't defined, it just evaluates to the double-wrapped parentheses which triggers the warning.

Could we just drop the parens? As is, even if the macro evaluates to `true`, the parentheses are not required; it could be rewritten as:

```
  if (alpha[0] == ZERO
#if defined(COMPLEX) && !defined(HERK)
      && alpha[1] == ZERO
#endif
      ) return 0;
```

which would evaluate just fine to ` if (alpha[0] == ZERO && alpha[1] == ZERO)`.
What do you think?

**Comment by NHDaly** [comment]
(I'd be willing to write and send a PR removing all those parens if you're amenable! 😸 )

**Comment by martin-frbg** [comment]
I may be missing something here, but I am not sure "we" would gain anything from such a change besides  removal of build log noise from the use of a specific compiler (clang ?) option that is not our suggested default ?

**Comment by brada4** [comment]
Proofreading out entire class of warnings one may find an inconsistency or a bug buried between next lines, then whole venture pays off on the spot.
In ideal world 'make -s' output fits in a screen or so...

**Comment by martin-frbg** [comment]
I'm all for builds that do not yield any compiler warning, but there are a number of them for bog-standard "-Wall" currently and in this particular case I expect we would need to run the "fix" past a few compilers on different platforms just to see that we do not get any new "missing parentheses" warnings or worse. Also I think wernsaar is or was working on just the affected files (which happen to be implicated in some of the recent threading-related bug reports)

**Comment by martin-frbg** [comment]
This was fixed by PR #1410 a few weeks ago.
